### ISSUE TITLE
Set meta description per-page

### ISSUE THREAD

**Issue by avivace** [gbdev/pandocs#393]
Currently, the description meta tag is set to a site-wide value, on every single exported HTML page:

```
<meta name="description" content="The single, most comprehensive technical reference to Game Boy available to the public.">
```

We need a proper couple of sentences, different for each page, to actually showcase what the page is about.

A couple of possible solutions:
   - Set description manually, per-page. This is the best as we can prepare curated "extracts". I'd like to be able to set it per-markdown file, maybe in a YAML front matter. See https://docs.rs/mdbook-frontmatter/0.0.4/mdbook_frontmatter/ can be useful
   - Unstrip the `{{{ content }}}` value of the HTML, ignore the headers and consider the first N (30, 40?) characters. Maybe even try to detect the first M (2, 3?) sentences.

**Comment by ISSOtm** [comment]
Imo, an automated solution would fall flat too often to be worth the effort. A per-file override would work much better imo.

**Comment by avivace** [comment]
> Imo, an automated solution would fall flat too often to be worth the effort. A per-file override would work much better imo.

Per-file override would be better, yes, but don't underestimate taking the first characters of the content. If you check https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fgbdev.io%2Fpandocs most of the autogenerated descriptions make sense (and they are just the first N words..)

**Comment by ISSOtm** [comment]
I agree they do make sense. However, I was thinking that they don't make the most sense as general descriptions of their pages' contents, rather most of them dive directly into the nitty-gritty. No idea if that's better or worse.
